docs/configuration.md

```diff
@@ -0,0 +1,210 @@
+
+Supported severity levels are:
+
+- `off`
+- `warn`
+- `error`
+
+You can define a severity level directly on the rule:
+
+Eg: `'no-bare-strings': 'warn'`
+
+OR
+
+If your rule has a custom configuration, and you want to define the severity level you would need to add the `severity` as the first element of the array.
+
+Eg:
+
+```js
+{
+   "no-implicit-this": ['warn', { "allow": [ "fooData" ] }
+}
+```
+
+## Per Template File
+
+It is also possible to disable specific rules (or all rules) in a template itself:
+
+```hbs
+<!-- disable all rules -->
+{{!-- template-lint-disable  --}}
+
+<!-- disable no-bare-strings -->
+{{!-- template-lint-disable no-bare-strings  --}}
+
+<!-- disable no-bare-strings and no-triple-curlies -->
+{{!-- template-lint-disable no-bare-strings no-triple-curlies  --}}
+
+<!-- enable all rules -->
+{{!-- template-lint-enable  --}}
+
+<!-- enable no-bare-strings -->
+{{!-- template-lint-enable no-bare-strings  --}}
+
+<!-- enable no-bare-strings and no-triple-curlies -->
+{{!-- template-lint-enable no-bare-strings no-triple-curlies  --}}
+```
+
+and to configure rules in the template:
+
+```hbs
+{{!-- template-lint-configure no-bare-strings ["ZOMG THIS IS ALLOWED!!!!"]  --}}
+
+{{!-- template-lint-configure no-bare-strings {"allowlist": "(),.", "globalAttributes": ["title"]}  --}}
+
+{{!-- template-lint-configure no-bare-strings ["warn", ["ZOMG THIS IS ALLOWED!!!!"]]  --}}
+
+{{!-- template-lint-configure no-bare-strings "warn"  --}}
+
+```
+
+The configure instruction can only configure a single rule, and the configuration value must be valid JSON that parses into a configuration for that rule.
+
+These configuration instructions do not modify the rule for the rest of the template, but instead modify it within whatever DOM scope the comment instruction appears.
+
+An instruction will apply to all later siblings and their descendants:
+
+```hbs
+<!-- disable for <p> and <span> and their contents, but not for <div> or <hr> -->
+<div>
+  <hr>
+  {{!-- template-lint-disable  --}}
+  <p>
+    <span>Hello!</span>
+  </p>
+</div>
+```
+
+An in-element instruction will apply to only that element:
+
+```hbs
+<!-- enable for <p>, but not for <div>, <hr> or <span> -->
+<div>
+  <hr>
+  <p {{!-- template-lint-enable  --}}>
+    <span>Hello!</span>
+  </p>
+</div>
+```
+
+An in-element instruction with the `-tree` suffix will apply to that element and all its descendants:
+
+```hbs
+<!-- configure for <p>, <span> and their contents, but not for <div> or <hr> -->
+<div>
+  <hr>
+  <p {{!-- template-lint-configure-tree block-indentation "tab"  --}}>
+    <span>Hello!</span>
+  </p>
+</div>
+```
+
+Note that enabling a rule (`{{!-- template-lint-enable --}}`) that has been configured in-template (`{{!-- template-lint-configure --}}`), will restore it to its default configuration rather than the modified in-template configuration for the scope of the `{{!-- template-lint-enable --}}` instruction.
+
+## Defining your own rules
```

Comment:
This plugin section with a link to the plugin doc should probably go in the top-level README.

Reply:
Seems fine by me.